### PR TITLE
mutex cleanup plus shared_ptr initialization loop

### PR DESCRIPTION
**Pull request by namccart**

### CODE REVIEW

File: include/es/pooled_resource.h [hunk 1]
```diff
@@ -20,10 +20,6 @@
 template <class T>
 class pooled_resource {
  public:
-    ~pooled_resource()
-        {
```
Comment (osh) [inline include/es/pooled_resource.h]: was this causing any problems? curious why you decided to remove it ?

File: include/es/pooled_resource.h [hunk 2]
```diff
@@ -9,20 +9,24 @@
...
+/*
  * this class pools a single resource type
  * providing the items from the pool when available
  * and NULL if all the pool items have been claimed
  * it does no allocation for you
  */
 template <class T>
 class pooled_resource {
-    public:
+ public:
```
Comment (osh) [inline include/es/pooled_resource.h]: wait you added this ... ? :-| ?

File: include/es/pooled_resource.h [hunk 3]
```diff
@@ -63,15 +67,13 @@ class managed_resource_pool {
     d_max(max_size)
     {
         ensure_allocated(pre_alloc_list);
-        _map_lock.unlock();
     }
  managed_resource_pool(boost::function<boost::shared_ptr<T> (IDX) > factory, int max_size=8, int initial_size=0, std::vector<IDX> pre_alloc_list = std::vector<IDX>() ) :
     d_factory(factory),
-    d_initial(initial_size),     
+    d_initial(initial_size),
```
Comment (osh) [inline include/es/pooled_resource.h]: handy whitespace adjustments

File: include/es/pooled_resource.h [hunk 4]
```diff
@@ -82,8 +84,12 @@ class managed_resource_pool {
         if(d_map.find(idx) == d_map.end()){
             boost::mutex::scoped_lock(_map_lock);
             d_map[idx] = boost::shared_ptr< pooled_resource <T > >(new pooled_resource<T>(d_max) );
+            while(d_map[idx].get() == 0)
```
Comment (osh) [inline include/es/pooled_resource.h]: would be nice to comment on this -- looks like we are busy looping until we can acquire one to populate --

File: include/es/pooled_resource.h (hunk 1 above)
Comment (namccart) [inline include/es/pooled_resource.h]: I'm pretty sure that I added that destructor yesterday when I was trying to kill a segfault.... I don't think it was doing anything useful, so it seemed better to put it back the way it was.

File: include/es/pooled_resource.h (hunk 2 above)
Comment (namccart) [inline include/es/pooled_resource.h]: this (the indenting) was just sloppiness... it's just where my editor auto-tabbed the public... not important.  I think it just auto-tabbed when I was messing around with the destructor.

File: include/es/pooled_resource.h (hunk 3 above)
Comment (namccart) [inline include/es/pooled_resource.h]: Tom's emacs settings are useful...

**Pull request merged**